### PR TITLE
Run basic predicates first in every check

### PR DESCRIPTION
**Pull request by MaciekPytel**
Previously CA run predicates in random order, this change makes it
run them starting with relatively cheap and most likely to fail
predicates, to avoid running more complex ones if they're not required.

### CODE REVIEW

File: cluster-autoscaler/simulator/predicates.go [hunk 1]
```diff
@@ -68,10 +78,27 @@ func NewPredicateChecker(kubeClient kube_client.Interface, stop <-chan struct{})
...
+			predicateList = append(predicateList, predicateInfo{name: predicateName, predicate: predicate})
+			delete(predicates, predicateName)
+		}
+	}
+	for predicateName, predicate := range predicates {
+		predicateList = append(predicateList, predicateInfo{name: predicateName, predicate: predicate})
+	}
+
+	for _, predInfo := range predicateList {
+		glog.Infof("Using predicate %s", predInfo.name)
```
Comment (mwielgus) [inline cluster-autoscaler/simulator/predicates.go]: V(1)

Comment (MaciekPytel) [inline cluster-autoscaler/simulator/predicates.go]: Done

**Comment by MaciekPytel** [comment]
@mwielgus Added PodFitsResources as a separate predicate as discussed.

**Pull request merged**